src/e2e-test/java/io/cdap/plugin/bigquerymultitable/stepdesign/BQMultiTableValidation.java

```diff
@@ -0,0 +1,230 @@
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * BQMTMultiTable Validation.
+ */
+
+public class BQMultiTableValidation {
+
+  private static final String tabA = "tabA";
+
+  private static final String tabB = "tabB";
+  public static Connection getMysqlConnection() throws SQLException, ClassNotFoundException {
+    String database = PluginPropertyUtils.pluginProp("databaseName");
+    Class.forName("com.mysql.cj.jdbc.Driver");
+    Connection connection = DriverManager.getConnection("jdbc:mysql://" + System.getenv("MYSQL_HOST") + ":" +
+                                     System.getenv("MYSQL_PORT") + "/" + database + "?tinyInt1isBit=false",
+                                     System.getenv("MYSQL_USERNAME"), System.getenv("MYSQL_PASSWORD"));
+    return connection;
+  }
+
+  /**
+   * Validates data from a MySQL table against corresponding BigQuery tables.
+   *
+   * @param sourceTable The name of the MySQL table to be validated.
+   * @return true if validation passes for all tables, false otherwise.
+   * @throws IOException If an I/O error occurs.
+   * @throws InterruptedException If the current thread is interrupted while waiting.
+   * @throws SQLException If an SQL error occurs.
+   * @throws ClassNotFoundException If the class specified is not found.
+   */
+
+  public static boolean validateMySqlToBQRecordValues(String sourceTable) throws IOException, InterruptedException,
+    SQLException, ClassNotFoundException {
+    // using MySql database and tables in Multiple Database table plugin
+    List<String> targetTables = getTableNameFromMySQL();
+    List<String> sourceList = new ArrayList<>();
+    sourceList.add(sourceTable);
+    List<Object> bigQueryRows = new ArrayList<>();
+    List<JsonObject> bigQueryResponse = new ArrayList<>();
+
+    Gson gson = new Gson();
+    for (int objNameIndex = 0; objNameIndex <= sourceList.size(); objNameIndex++) {
+      String currentTargetTable = targetTables.get(objNameIndex);
+      getBigQueryTableData(currentTargetTable, bigQueryRows);
+    }
+    for (Object row : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    bigQueryResponse.sort(Comparator.comparing(jsonData -> jsonData.get("EmployeeID").getAsInt()));
+
+
+    String getTargetQuery = "SELECT * FROM "  + sourceTable;
+    try (Connection connect = getMysqlConnection()) {
+      connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      Statement statement1 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE,
+                                                     ResultSet.CONCUR_UPDATABLE, ResultSet.HOLD_CURSORS_OVER_COMMIT);
```

Comment:
Setting Result Set holdability for the sql statement is redundant in my opinion since you already set it in the MySql connection